### PR TITLE
community/dnscrypt-proxy update to version 2.0.4

### PR DESCRIPTION
**Pull request by ianbashford**
Updated to build v2.0.4
Setup package removed as config file has changed
Init.d script updated. Config moved to /etc/dnscrypt-proxy
Starts with the example config file which has sensible defaults

### CODE REVIEW

**Comment by nmeum** [comment]
Could you split the upgrade and all the other changes you made into separate commits?

**Comment by ianbashford** [comment]
FYI all the changes are related.  
The upgrade to 2.x is actually a rewrite by upstream, so the other changes in this commit (to init.d and the pre/post install scripts)  and the deletions (old config file and a setup script for generating the old style config file)  are all required by the upgrade.
Are you OK with commits that rewrite history, or prefer a revert and then new piecemeal commits?

**Comment by itoffshore** [comment]
version `2.0.7` is out now

**Comment by ianbashford** [comment]
Done.
I've also added a post-upgrade and pre-deinstall script to do some tidying
up.

I've not renamed the branch however, as I believe that'll close the pull
request...

**Comment by andypost** [comment]
Great rework! I suggest you to squash all commits using interactive rebase https://github.com/alpinelinux/aports/blob/master/.github/CONTRIBUTING.md#clean-up-a-pull-request-pr

**Comment by itoffshore** [comment]
@ianbashford - I also have [some notes here for rebasing](https://it-offshore.co.uk/code/81-alpine-linux-pull-requests-on-github)

**Comment by ianbashford** [comment]
Thanks for the links to the notes @itoffshore and @andypost

File: community/dnscrypt-proxy/dnscrypt-proxy.pre-deinstall [hunk 1]
```diff
@@ -0,0 +1,5 @@
+#!/bin/sh
+
+/etc/init.d/dnscrypt-proxy stop
+
+exit 0
```
Comment (jirutka) [inline community/dnscrypt-proxy/dnscrypt-proxy.pre-deinstall]: This should not be here. It’s admin’s responsibility to stop the service. Please remove it.

File: community/dnscrypt-proxy/dnscrypt-proxy.post-install [hunk 2]
```diff
@@ -0,0 +1,5 @@
+#!/bin/sh
+
+setcap cap_net_bind_service=+pe /usr/sbin/dnscrypt-proxy
```
Comment (jirutka) [inline community/dnscrypt-proxy/dnscrypt-proxy.post-install]: This should be done in APKBUILD (in package phase), not in post-install script.

File: community/dnscrypt-proxy/dnscrypt-proxy.initd [hunk 3]
```diff
@@ -1,24 +1,17 @@
...
-command=/usr/sbin/dnscrypt-proxy
-command_args="--pidfile=$pidfile --daemonize
-	--logfile=${DNSCRYPT_LOGFILE:-/var/log/dnscrypt-proxy/dnscrypt-proxy.log}
-	--user=${DNSCRYPT_USER:-dnscrypt}
-	--local-address=${DNSCRYPT_LOCALIP:-127.0.0.1:53}"
-
+command="/usr/sbin/dnscrypt-proxy"
+command_args="-config /etc/dnscrypt-proxy/dnscrypt-proxy.toml -syslog"
+command_user="${DNSCRYPT_USER:-dnscrypt}:${DNSCRYPT_GROUP:-dnscrypt}"
+pidfile="/var/run/${SVCNAME}.pid"
```
Comment (jirutka) [inline community/dnscrypt-proxy/dnscrypt-proxy.initd]: `/var/run` is legacy, pid files are in `/run`.

File: community/dnscrypt-proxy/dnscrypt-proxy.initd [hunk 4]
```diff
@@ -1,24 +1,17 @@
...
+start_stop_daemon_args="--background --make-pidfile"
```
Comment (jirutka) [inline community/dnscrypt-proxy/dnscrypt-proxy.initd]: Please define variable `command_background="yes"` instead.

File: community/dnscrypt-proxy/dnscrypt-proxy.initd (hunk 3 above)
Comment (jirutka) [reply community/dnscrypt-proxy/dnscrypt-proxy.initd]: `$SVCNAME` is also legacy, use `$RC_SVCNAME`.

File: community/dnscrypt-proxy/dnscrypt-proxy.initd [hunk 5]
```diff
@@ -1,24 +1,17 @@
...
+    provide dns
 }
 
 start_pre() {
-	checkpath --directory ${pidfile%/*}
-	# by default opendns is used but its possible to override
-	if [ -n "$RESOLVER" ] && [ -n "$PUBKEY" ] && [ -n "$PROVIDER" ]; then
-		command_args="$command_args -r ${RESOLVER} -k ${PUBKEY} -N ${PROVIDER}"
-	fi
+    #checkpath -q -d -m 0775 -o "${command_user}" /var/cache/"${SVCNAME}"
```
Comment (jirutka) [inline community/dnscrypt-proxy/dnscrypt-proxy.initd]: Why is it here?

File: community/dnscrypt-proxy/APKBUILD [hunk 6]
```diff
@@ -1,62 +1,50 @@
...
-pkgrel=2
+pkgver=2.0.8
+pkgrel=1
 pkgdesc="A tool for securing communications between a client and a DNS resolver"
 url="https://dnscrypt.org/"
 arch="all"
 license="custom"
-makedepends="$depends_dev libsodium-dev ldns-dev"
-install="$pkgname.pre-install"
+makedepends="$depends_dev go"
```
Comment (jirutka) [inline community/dnscrypt-proxy/APKBUILD]: Variable `depends_dev` is not define here, remove it.

File: community/dnscrypt-proxy/APKBUILD [hunk 7]
```diff
@@ -1,62 +1,50 @@
...
-		--build=$CBUILD \
-		--host=$CHOST \
-		--prefix=/usr
-	make
+        cd "$srcdir"
+        mv $pkgname-$pkgver src
+        export GOPATH=$PWD
+	cd src/$pkgname
+        go clean
+        go build -ldflags="-s -w" -o $GOPATH/../linux-amd64/dnscrypt-proxy
```
Comment (jirutka) [inline community/dnscrypt-proxy/APKBUILD]: Don’t mix tabs and spaces, use only tabs for indentation.

Comment (jirutka) [reply community/dnscrypt-proxy/APKBUILD]: > `linux-amd64`

We support more architectures than just x86_64 (amd64).

**Pull request closed without merging**